### PR TITLE
refactor: reduce validation code public API

### PR DESCRIPTION
**Pull request by phisco**
### Description of your changes

As suggested I'm reducing the public API for the schema validation code introduced by #3937.
I'm also renaming a few tests according to the convention as suggested #3983.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [x] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

No need, just a refactor, unit tests are passing.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Review by negz — APPROVED** [review]
Thanks! I'm definitely open to moving these back out, but in general I like to start with everything in `internal`.

File: apis/apiextensions/v1/composition_patches.go [hunk 1]
```diff
@@ -23,7 +23,7 @@ import (
 
 	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
 
-	"github.com/crossplane/crossplane/pkg/validation/errors"
+	"github.com/crossplane/crossplane/internal/errors"
```
Comment (negz) [inline apis/apiextensions/v1/composition_patches.go:26]: Maybe `internal/validation/errors` for this package? That way its less likely to get confused with the existing general error-handling package in crossplane-runtime.

File: apis/apiextensions/v1/composition_transforms.go [hunk 2]
```diff
@@ -27,8 +27,7 @@ import (
 
 	"github.com/crossplane/crossplane-runtime/pkg/errors"
 
-	xperrors "github.com/crossplane/crossplane/pkg/validation/errors"
-	"github.com/crossplane/crossplane/pkg/validation/schema"
+	xperrors "github.com/crossplane/crossplane/internal/errors"
```
Comment (negz) [inline apis/apiextensions/v1/composition_transforms.go:30]: Nit: Maybe `verrors` or `validationerrors`? To distinguish that these errors pertain to validation, not generic "crossplane" errors as is implied by the prefix "xp".

Comment (phisco) [reply apis/apiextensions/v1/composition_transforms.go]: I went with `verrors` https://github.com/crossplane/crossplane/pull/3986/commits/62ec5ba5eaa314385734434d46e2dd94bd1515fb

File: apis/apiextensions/v1/composition_patches.go (hunk 1 above)
Comment (phisco) [reply apis/apiextensions/v1/composition_patches.go]: moved 👍

**Review by negz — APPROVED** [review]

**Pull request merged**